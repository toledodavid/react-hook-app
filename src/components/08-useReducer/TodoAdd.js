import React from 'react';
import { useForm } from '../../hooks/useForm';



const TodoAdd = ({handleAddTodo}) => {

  const [{description}, handleInputChange, reset] = useForm({
    description: ''
  });

  const handleSubmit = (event) => {
    event.preventDefault();

    if (description.trim().length <= 1) {
      return;
    }

    const newTodo = {
      id: new Date().getTime(),
      desc: description,
      done: false
    }

    handleAddTodo(newTodo);
    reset();
  }


  return(
    <>
      <h4>Add TODO</h4>
      <hr/>

      <form onSubmit={handleSubmit}>
        <input type="text" 
                name="description"
                placeholder="Learn..."
                autoComplete="off"
                className="form-control"
                value={description}
                onChange={handleInputChange} />

        <button type="submit" className="btn btn-outline-primary btn-block mt-1">Add</button>
      </form>
    </>
  );
}


export default TodoAdd;