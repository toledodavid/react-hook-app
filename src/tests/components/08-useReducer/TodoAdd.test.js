import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import TodoAdd from '../../../components/08-useReducer/TodoAdd';


describe('Tests in <TodoAdd /> component', () => {

  const handleAddTodo = jest.fn();

  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);

  test('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('It should NOT call handleAddTodo function', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({preventDefault(){}});

    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test('It should call handleAddTodo function', () => {
    const value = 'Learn Node';

    wrapper.find('input').simulate('change', {
      target: {
        value,
        name: 'description'
      }
    });

    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({preventDefault(){}});

    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
    expect(handleAddTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      desc: value,
      done: false
    });
    expect(wrapper.find('input').prop('value')).toBe('');
  });

});