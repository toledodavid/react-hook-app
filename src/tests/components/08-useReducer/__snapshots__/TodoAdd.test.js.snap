// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <TodoAdd /> component It should render correctly 1`] = `
<Fragment>
  <h4>
    Add TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Learn..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary btn-block mt-1"
      type="submit"
    >
      Add
    </button>
  </form>
</Fragment>
`;
